src/modalities/models/huggingface_adapters/hf_adapter.py

```diff
@@ -0,0 +1,93 @@
+import json
+from dataclasses import dataclass
+from pathlib import PosixPath
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+from transformers import PreTrainedModel, PretrainedConfig
+from transformers.utils import ModelOutput
+
+from modalities.models.model import NNModel
+from modalities.models.utils import get_model_from_config
+
+
+class HFModelAdapterConfig(PretrainedConfig):
+    model_type = "modalities"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # self.config is added by the super class via kwargs
+        assert self.config, "config is not set"
+        # since the config will be saved to json and json can't handle posixpaths, we need to convert them to strings
+        if self.config:
+            self._convert_posixpath_to_str(self.config)
+
+    def to_json_string(self, use_diff: bool = True) -> str:
+        if self.config:
+            json_dict = {"config": self.config.copy(), "model_type": self.model_type}
+        else:
+            json_dict = {}
+        return json.dumps(json_dict)
+
+    def _convert_posixpath_to_str(self, d: dict):
+        """
+        Recursively iterate over the dictionary and convert PosixPath values to strings.
+        """
+        for key, value in d.items():
+            if isinstance(value, PosixPath):
+                d[key] = str(value)
+            elif isinstance(value, dict):
+                self._convert_posixpath_to_str(value)
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, PosixPath):
+                        d[key] = str(item)
```

Comment:
I think this is a bug. 
here we replace a list of Posixpaths and posibly other items to a single posixpath string. Is this wanted behavior?

Reply:
We should cover this method with unit testing.